crates/ratchet-core/src/gpu/wgsl/dtype.rs

```diff
@@ -0,0 +1,35 @@
+use half::f16;
+
+/// Supported data types in WGSL.
+///
+/// This can be mapped to and from the Ratchet DType.
+
+pub trait WgslDType {
+    fn render_dt() -> &'static str {
+        unimplemented!()
+    }
```

Comment:
If they are always as simple as below you could go for an associated constant
```suggestion
    const DT: &'static str;
```

Reply:
Good idea, I should have thought of this.

Reply:
You eventually would have ;)